### DISCUSSION TITLE
Create proxy when deserialize

### DISCUSSION THREAD

**Discussion by Luigi6821** [dotnet/runtime#83504] [General]
Hi I am facing how to create a proxy instance of a class that I want deserialize.
Using Newtonsoft I can do the following:
```C#
public class ProxyConverter<T> : Newtonsoft.Json.Converters.CustomCreationConverter<T>
    where T : BaseClass
{
    static private ProxyGenerator _generator = new ProxyGenerator();

    public override T Create(Type objectType)
    {
        var proxy = _generator.CreateClassProxy<T>(new CallInterceptor());
        return proxy;
    }

    public static Newtonsoft.Json.JsonConverter CreateInstance<CO, CL>()
        where CO : NSProxyConverter<CL>
        where CL : myParentClass
    {
        return new NSProxyConverter<CL>();
    }
} 

var proxy = Newtonsoft.Json.JsonConvert.DeserializeObject<myClass>("{\"Id\":1.0}", new ProxyConverter<myClass>());
```

In the above example I create an instance of a proxy of class T (inherited from BaseClass)
ProxyGenerator is from pacakge Castle.Core
How can do same with System.Text json classes?

Thanks in advance
Luigi

**Comment by eiriktsarpalis** [comment]
Hi, there's no equivalent to `CustomCreationConverter` in System.Text.Json. You should be able to achieve similar results using the `JsonTypeInfo.CreateObject` delegate in the [contract customization layer](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/custom-contracts).
